index.d.ts

```diff
@@ -14,6 +14,7 @@ export interface AxiosBasicCredentials {
 export interface AxiosProxyConfig {
   host: string;
   port: number;
+  isHttps?: boolean;
```

Comment:
I'd use a `protocol` option instead, accepting both `http` and `http:` formats so parsed URL can be passed as the proxy option.